src/aind_behavior_curriculum/tmp.py

```diff
@@ -0,0 +1,80 @@
+"""
+Work in Progress Curriculum
+"""
+
+from collections import defaultdict
+
+class Task:
+    """
+    Defined in exisiting PR
+    """
+    pass
+
+class Metric:
+    """
+    Initalizes a connection to relevant database (referencing some configuration file)
+    User extends this with their own metric implementation based on data inside the database.
+    """
+    pass
+
+
+class Evaluation:
+    """
+    Serializable Callable, takes in a list of metrics that is serialized alongside.
+    """
+    pass
+
+
+# TODO: De/serialization
+class Curriculum:
```

Comment:
We might also need a "finally" -like evaluation that runs if no transition evaluates to true. This clause is responsible for updating the current stage settings instead of transitioning the animal. Should probably be a dict[stage, eval] since there should only be one per stage.